frontend-v2/src/components/nav.tsx

```diff
@@ -0,0 +1,148 @@
+  session: Awaited<ReturnType<typeof fetchSession>>
+}) => {
+  const [isMobileExpanded, setMobileExpanded] = useState(false)
+  const [activeDropdown, setActiveDropdown] = useState<string | null>(null)
+
+  const hasAccess = useCallback(
+    (roleRequired: string[] | undefined) => {
+      if (!roleRequired) return true
+
+      const role = session?.user?.role
+      if (!role) return false
+
+      return roleRequired.some((it) => {
+        if (it === 'admin') return role === 'admin'
+        if (it === 'organizer') return role === 'admin' || role === 'organizer'
+        if (it === 'attendance')
+          return (
+            role === 'admin' || role === 'organizer' || role === 'attendance'
+          )
+        if (it === 'non-sfbay') return role === 'non-sfbay'
+        return false
+      })
+    },
+    [session?.user?.role],
+  )
+
+  // Note that this navbar currently uses the Vue stylesheet. Once we
+  // are no longer using Vue, we should update this using tailwind.
+  return (
+    <nav
+      role="navigation"
+      aria-label="main navigation"
+      className="navbar is-fixed-top has-shadow"
+      id="mainNav"
+    >
+      <div className="navbar-brand">
+        <div className="navbar-item">
+          <Image src={logo1} alt="DxE" className="w-auto h-auto" priority />
+        </div>
+        <button
+          aria-label="menu"
+          className={cn('navbar-burger burger', {
+            'is-active': isMobileExpanded,
+          })}
+          onClick={() => setMobileExpanded((prev) => !prev)}
+        >
+          <span aria-hidden="true"></span>
+          <span aria-hidden="true"></span>
+          <span aria-hidden="true"></span>
+        </button>
+      </div>
+
+      <div className={cn('navbar-menu', { 'is-active': isMobileExpanded })}>
+        <div className="navbar-start">
+          {navbarData.items.map(
+            (dropdown) =>
+              hasAccess(dropdown.roleRequired) && (
+                <div className="navbar-item has-dropdown" key={dropdown.label}>
+                  {/* Important: This must remain an `a` element and not a `button` for the buefy mobile styles to work properly. */}
+                  <a
+                    role="menuitem"
+                    aria-haspopup
+                    className="navbar-link"
+                    onClick={(e) => {
+                      e.preventDefault()
+                      setActiveDropdown((prev) =>
+                        prev === dropdown.label ? null : dropdown.label,
+                      )
+                    }}
+                  >
+                    {dropdown.label}
+                  </a>
+                  {activeDropdown === dropdown.label && (
+                    <div
+                      className={cn('navbar-dropdown !block')}
+                      onClick={() => setActiveDropdown(dropdown.label)}
+                    >
+                      {dropdown.items.map((item) => {
```

Comment:
could the dropdown items be extracted to a variable just before the return statement? i think it would make it easier to follow the flow of the remaining JSX  / see the overall outline of the component, and help reduce such deep nesting.

Reply:
sure, lemme refactor some stuff a bit so we don't have to drill everything so much